diff -uNr xcin.org/cin/big5/Makefile.in xcin/cin/big5/Makefile.in
--- xcin.org/cin/big5/Makefile.in	2001-09-06 23:41:32.000000000 +0800
+++ xcin/cin/big5/Makefile.in	2003-12-23 10:36:32.000000000 +0800
@@ -9,7 +9,7 @@
 all: all-$(OS_TYPE)
 
 all-LINUX:
-	( BIG5ENC=big5 BIG5HKENC=big5hkscs make -f Rules make-big5 make-big5hkscs )
+	( BIG5ENC=big5 BIG5HKENC=big5hkscs UTF8ENC=utf-8 make -f Rules make-big5 make-big5hkscs make-utf8 )
 
 all-FREEBSD:
 	( BIG5ENC=big5 make -f Rules make-big5 )
@@ -30,7 +30,7 @@
 install: all install-$(OS_TYPE)
 
 install-LINUX:
-	( BIG5ENC=big5 BIG5HKENC=big5hkscs make -f Rules install_big5 install_big5hkscs )
+	( BIG5ENC=big5 BIG5HKENC=big5hkscs UTF8ENC=utf-8 make -f Rules install_big5 install_big5hkscs install_utf8 )
 
 install-FREEBSD:
 	( BIG5ENC=big5 make -f Rules install_big5 )
diff -uNr xcin.org/cin/big5/Rules xcin/cin/big5/Rules
--- xcin.org/cin/big5/Rules	2001-11-28 22:40:39.000000000 +0800
+++ xcin/cin/big5/Rules	2003-12-23 10:37:37.000000000 +0800
@@ -21,12 +21,27 @@
 		jyutping0.tab.$(BIG5HKENC) \
 		telecode.tab.$(BIG5HKENC)
 
+utf-8 = 	sys.tab.$(UTF8ENC) \
+		array30.tab.$(UTF8ENC) \
+		cj.tab.$(UTF8ENC) \
+		cj5.tab.$(UTF8ENC) \
+		phone.tab.$(UTF8ENC) \
+		pinyin.tab.$(UTF8ENC) \
+		pinyin0.tab.$(UTF8ENC) \
+		simplex.tab.$(UTF8ENC) \
+		simplex5.tab.$(UTF8ENC) \
+		jyutping.tab.$(UTF8ENC) \
+		jyutping0.tab.$(UTF8ENC) \
+		telecode.tab.$(UTF8ENC) \
+		bims_pinyin.tab.$(UTF8ENC) \
+		default.phr.$(UTF8ENC)
 include Makefile
 
 make-big5: $(big5) 
 
 make-big5hkscs: $(big5hkscs)
 
+make-utf8: $(utf-8)
 
 sys.tab.$(BIG5ENC): sys.cin.big5
 	$(build) -r $(xcinrc) -l $(BIG5ENC) -o $@ sys.cin.big5
@@ -91,6 +106,50 @@
 	$(build) -r $(xcinrc) -l $(BIG5HKENC) \
 		-s sys.tab.$(BIG5HKENC) -o $@ telecode.cin
 
+sys.tab.$(UTF8ENC): sys.cin.utf-8
+	$(build) -r $(xcinrc) -l $(UTF8ENC) -o $@ sys.cin.utf-8
+	for f in *.cin ; do \
+	    iconv --verbose -c -f BIG5HKSCS -t UTF-8 $$f -o $$f.$(UTF8ENC) ;\
+	done
+	cp -f bims_pinyin.cin bims_pinyin.cin.utf-8
+array30.tab.$(UTF8ENC): array30.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ array30.cin.$(UTF8ENC)
+cj.tab.$(UTF8ENC): cj.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ cj.cin.$(UTF8ENC)
+cj5.tab.$(UTF8ENC): cj5.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ cj5.cin.$(UTF8ENC)
+phone.tab.$(UTF8ENC): phone.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ phone.cin.$(UTF8ENC)
+pinyin.tab.$(UTF8ENC): pinyin.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ pinyin.cin.$(UTF8ENC)
+pinyin0.tab.$(UTF8ENC): pinyin0.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ pinyin0.cin.$(UTF8ENC)
+simplex.tab.$(UTF8ENC): simplex.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ simplex.cin.$(UTF8ENC)
+simplex5.tab.$(UTF8ENC): simplex5.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ simplex5.cin.$(UTF8ENC)
+jyutping.tab.$(UTF8ENC): jyutping.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ jyutping.cin.$(UTF8ENC)
+jyutping0.tab.$(UTF8ENC): jyutping0.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ jyutping0.cin.$(UTF8ENC)
+telecode.tab.$(UTF8ENC): telecode.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ telecode.cin.$(UTF8ENC)
+bims_pinyin.tab.$(UTF8ENC): bims_pinyin.cin.$(UTF8ENC) sys.tab.$(UTF8ENC)
+	$(build) -r $(xcinrc) -l $(UTF8ENC) \
+		-s sys.tab.$(UTF8ENC) -o $@ bims_pinyin.cin.$(UTF8ENC)
+default.phr.$(UTF8ENC): default.phr
+	iconv -f big5 -t utf8 default.phr > default.phr.$(UTF8ENC)
 
 install_big5:
 	if [ ! -d $(xcin_datap)/$(BIG5ENC) ]; then \
@@ -106,3 +165,9 @@
 	  ff=`basename $$f .$(BIG5HKENC)`; \
 	  $(INSTALL_DATA) $$f $(xcin_datap)/$(BIG5HKENC)/$$ff; done
 
+install_utf8:
+	if [ ! -d $(xcin_datap)/$(UTF8ENC) ]; then \
+	  $(INSTALL) -d $(xcin_datap)/$(UTF8ENC); fi
+	for f in $(utf-8) ; do \
+	  ff=`basename $$f .$(UTF8ENC)`; \
+	  $(INSTALL_DATA) $$f $(xcin_datap)/$(UTF8ENC)/$$ff; done
